sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -3104,10 +3104,10 @@ def hpat_pandas_series_argsort_idx_impl(self, axis=0, kind='quicksort', order=No
                 sort_nona = numpy.argsort(self._data[~na_data_arr])
             q = 0
             for id, i in enumerate(sort):
-                if id not in list(sort[len(self._data) - na:]):
-                    result[id] = sort_nona[id-q]
-                else:
+                if id in list(sort[len(self._data) - na:]):
```

Comment:
Not the best name `sort` for the Python variable. Why do you use constructor `list()` here? Is it exactly required?